src/parser.rs

```diff
@@ -537,34 +574,58 @@ impl Library {
                             ))
                         }
                         "record" => {
-                            #[cfg(feature = "use_unions")]
-                            {
-                                let mut r = match try!(self.read_record(parser, ns_id, attrs)) {
-                                    Some(Type::Record(r)) => r,
-                                    _ => continue,
-                                };
+                            let mut r = match try!(self.read_record(
+                                parser,
+                                ns_id,
+                                attrs,
+                                parent_name_prefix,
+                                parent_ctype_prefix
+                            )) {
+                                Some(Type::Record(r)) => r,
+                                _ => continue,
+                            };
+
+                            let field_name = r.name.clone();
```

Comment:
```rust
                            let mut field_name = String::new();
                            let mut type_name = String::new();
                            for attr in &attributes {
                                match attr.name.local_name.as_ref() {
                                    "name" => field_name = attr.value.clone(),
                                    "type" => type_name = attr.value.clone(),
                                    _ => {},
                                }
                            }
```
This appears to do the job nicely. Can probably do without the `type_name` though, I don't know where that would apply tbh.

Can now also adjust;
```rust
                                u = Union {
                                    name: format!(
                                        "{}{}_{}{}",
                                        parent_name_prefix.unwrap_or(""),
                                        record_name,
                                        field_name,
                                        union_count
                                    ),
                                    c_type: Some(format!(
                                        "{}{}_{}{}",
                                        parent_ctype_prefix.unwrap_or(""),
                                        c_type,
                                        field_name,
                                        union_count
                                    )),
                                    ..u
                                };
```